sable_ircd/src/command/handlers/mode.rs

```diff
@@ -210,7 +210,10 @@ async fn handle_channel_mode(
                         // Can't query keys
                         Direction::Query => (),
                         Direction::Add => {
-                            let new_key = args.next::<ChannelKey>()?;
+                            let new_key = match args.next().map(ChannelKey::new_coerce)? {
+                                Ok(key) => key,
+                                Err(_) => return numeric_error!(InvalidKey, &chan.name()),
```

Comment:
I favored [ERR_INVALIDKEY](https://modern.ircdocs.horse/#errinvalidkey-525) over [ERR_INVALIDMODEPARAM](https://modern.ircdocs.horse/#errinvalidmodeparam-696) here because it does not include a `key` parameter that we can't print (because the key is either empty or contains a space).